src/KernelAbstractions.jl

```diff
@@ -62,10 +67,63 @@ abstract type Event end
 import Base.wait
 
 # TODO
-function async_copy! end
 # function register end
 # function unregister end
 
+# Functions for async_copy
+function pin!(a)
```

Comment:
probably want a `pin!(::CUDA, a)` so that we can pin with respect to different backends.